code/src/pages/cart/CartPage.tsx

```diff
@@ -19,118 +22,120 @@ type CartItemsType = {
 };
 
 const CartPage = () => {
-  const [cartItems, setCartItems] = useState<CartItemsType[]>(items);
+  const { id: userId } = useUserDetailsSelector() || {};
+  const {
+    data: cartItems,
+    error,
+    isLoading: cartItemsLoading
+  } = useGetCartItemsQuery(userId!);
```

Comment:
This can be refactored like I wrote for @D1nviz's PR. Seems like he is not available for now. Maybe you will merge it faster, so you can also fix it. What do you think?

https://github.com/idx-academy/spa-orders/pull/258#discussion_r1675404438

Reply:
> This can be refactored like I wrote for @D1nviz's PR to avoid `|| {}` and `userId!`. Seems like he is not available for now. Maybe you will merge it faster, so you can also fix it. What do you think?
> 
> Link to comment: [#258 (comment)](https://github.com/idx-academy/spa-orders/pull/258#discussion_r1675404438)

He used `eslint disable`. In my case it should be baecause you know about marker `!` -> `!null && !undefined`

Reply:
Consider following example:

```jsx
 const { id: userId } = useUserDetailsSelector() || {};
```

You need `|| {}` part here because `useUserDetailsSelector`'s return type is `UserDetails | null`. It is related only to typescript part. You will not need those if `useUserDetailsSelector()` will never return `null`. Also it will not be required to do in other places through the code, that will drastically cleans up our codebase

Reply:
> Consider following example:
> 
> ```js
>  const { id: userId } = useUserDetailsSelector() || {};
> ```
> 
> You need `|| {}` part here because `useUserDetailsSelector`'s return type is `UserDetails | null`. It is related only to typescript part. You will not need those if `useUserDetailsSelector()` will never return `null`. Also it will not be required to do in other places through the code, that will drastically cleans up our codebase

As I talked. When we discussed with @D1nviz . For refactor should be another issue. We should move forward with tasks and not be busy with refactoring

Reply:
We don't need to check for authorization twice, because it is already performed by `ProtectedRoute` component. We should not ever call `useUserDetailsSelector` when we are not under `ProtectedRoute`. Do you agree with it?